### PR TITLE
[DirectX] Register a few DXIL passes with the new PM

### PR DESCRIPTION
**Pull request by bogner**
This wires up dxil-op-lower, dxil-intrinsic-expansion, dxil-translate-metadata,
and dxil-pretty-printer to the new pass manager, both as a matter of future
proofing the backend and so that they can be used more flexibly in tests.

A few arbitrary tests are updated in order to test the new PM path, and we drop
the "print-dxil-resource-md" pass since it's redundant with the pretty printer.

### CODE REVIEW

**Comment by bogner** [comment]
Depends on #104248

File: llvm/lib/Target/DirectX/DirectXPassRegistry.def [hunk 1]
```diff
@@ -23,7 +23,10 @@ MODULE_ANALYSIS("dxil-resource-md", DXILResourceMDAnalysis())
 #ifndef MODULE_PASS
 #define MODULE_PASS(NAME, CREATE_PASS)
 #endif
+MODULE_PASS("dxil-intrinsic-expansion", DXILIntrinsicExpansion())
+MODULE_PASS("dxil-op-lower", DXILOpLowering())
+MODULE_PASS("dxil-pretty-printer", DXILPrettyPrinterPass(dbgs()))
+MODULE_PASS("dxil-translate-metadata", DXILTranslateMetadata())
 // TODO: rename to print<foo> after NPM switch
 MODULE_PASS("print-dx-shader-flags", dxil::ShaderFlagsAnalysisPrinter(dbgs()))
-MODULE_PASS("print-dxil-resource-md", DXILResourceMDPrinterPass(dbgs()))
```
Comment (python3kgae) [inline llvm/lib/Target/DirectX/DirectXPassRegistry.def:28]: Should we remove DXILResourceMDPrinterPass in lib\Target\DirectX\DXILResourceAnalysis.cpp ?

Comment (bogner) [reply llvm/lib/Target/DirectX/DirectXPassRegistry.def:28]: Yes we should. I currently have that as part of this change that isn't quite ready for review yet: https://github.com/bogner/llvm-project/commit/bed151261c5b26519f6e44c44631cc1f770dfd78

Maybe I should pull the deletion into this PR instead?

Comment (bogner) [reply llvm/lib/Target/DirectX/DirectXPassRegistry.def:28]: I've pulled the removal of the pass into this change.

**Review by python3kgae — APPROVED** [review]

**Review by hekota — APPROVED** [review]
LGTM!

**Review by farzonl — APPROVED** [review]

File: llvm/test/Analysis/DXILResource/buffer-frombinding.ll [hunk 2]
```diff
@@ -46,14 +46,14 @@ define void @test_typedbuffer() {
   ; Buffer<uint4> Buf[24] : register(t3, space5)
   %typed2 = call target("dx.TypedBuffer", <4 x i32>, 0, 0, 0)
       @llvm.dx.handle.fromBinding.tdx.TypedBuffer_i32_0_0t(
-          i32 2, i32 7, i32 24, i32 0, i1 false)
+          i32 5, i32 3, i32 24, i32 0, i1 false)
```
Comment (damyanp) [inline llvm/test/Analysis/DXILResource/buffer-frombinding.ll:49]: What happened here?

**Review by damyanp — APPROVED** [review]

File: llvm/test/Analysis/DXILResource/buffer-frombinding.ll (hunk 2 above)
Comment (bogner) [reply llvm/test/Analysis/DXILResource/buffer-frombinding.ll:49]: That was supposed to be included in #100699 in response to [a comment](https://github.com/llvm/llvm-project/pull/100699#discussion_r1715835841), but I guess I somehow put it in the wrong change. Committing that separately here: #105409

**Pull request merged**